syntax = "proto2";

package com.google.tuningfork;

// Passed by the user to tuning fork at initialization.
message Settings {
  message Histogram {
    optional int32 instrument_key = 1;
    optional float bucket_min = 2;
    optional float bucket_max = 3;
    optional int32 n_buckets = 4;
  }
  message AggregationStrategy {
    enum Submission {
      INVALID = 0;
      TIME_BASED = 1;
      TICK_BASED = 2;
    }
    optional Submission method = 1;
    optional int32 intervalms_or_count = 2;
    optional int32 max_instrumentation_keys = 3;
    repeated int32 annotation_enum_size = 4;
  }
  optional AggregationStrategy aggregation_strategy = 1;
  repeated Histogram histograms = 2;
  optional string base_uri = 3;
  optional string api_key = 4;
  optional string default_fp_filename = 5;
  optional int32 initial_request_timeout_ms = 6;
  optional int32 ultimate_request_timeout_ms = 7;
  optional int32 loading_annotation_index = 8; // 1-based index
  optional int32 level_annotation_index = 9; // 1-based index
}
